crates/validation/src/header_validator.rs

```diff
@@ -96,37 +102,34 @@ impl HeaderValidator {
     ) -> anyhow::Result<()> {
         if block_number < MERGE_BLOCK_NUMBER {
             return Err(anyhow!(
-                "Invalid HistoricalRootsBlockProof found for pre-merge header."
+                "Invalid BlockProofHistoricalRoots found for pre-Merge header."
             ));
         }
         if block_number >= SHANGHAI_BLOCK_NUMBER {
             return Err(anyhow!(
-                "Invalid HistoricalRootsBlockProof found for post-Shanghai header."
+                "Invalid BlockProofHistoricalRoots found for post-Shanghai header."
             ));
         }
 
-        // Verify the chain of proofs for post-merge/pre-capella block header
-        Self::verify_bellatrix_to_deneb_beacon_block_proof(
+        // Verify the chain of proofs for execution block header inclusion in beacon block
+        if !self.verify_bellatrix_to_deneb_execution_block_proo(
             header_hash,
             &proof.execution_block_proof,
             proof.beacon_block_root,
-        )?;
-
-        let block_root_index = proof.slot % EPOCH_SIZE;
-        let gen_index = 2 * EPOCH_SIZE + block_root_index;
-        let historical_root_index = proof.slot / EPOCH_SIZE;
-        let historical_root =
-            self.historical_roots_acc.historical_roots[historical_root_index as usize];
+        ) {
+            return Err(anyhow!(
+                "Execution block proof verification failed for Merge-Capella header"
+            ));
```

Comment:
```suggestion
            bail!("Execution block proof verification failed for Merge-Capella header");
```

Couldn't we just use `bail!(` instead of `return Err(anyhow!(` everywhere?